python/ray/tune/trial_runner.py

```diff
@@ -1034,6 +1047,8 @@ def _update_trial_queue(self, blocking: bool = False,
         Returns:
             Boolean indicating if a new trial was created or not.
         """
+        self._updated_queue = True
```

Comment:
what does this variable `_updated_queue` represent? it seems like this function can return False while _updated_queue becomes set to True

Reply:
See line 469-471. We want to keep updating the true only if it was successful before or if we haven't tried. If it was unsuccessful (`self._updated_queue=True` and `next_trial=None`) we don't want to try again, as we already know it failed.

Btw line 471 reduces to `if not self._updated_queue or next_trial:`, but I might leave it as is for better verbosity